Try fixing travis:test pr

Tlazypanda opened the pull request:
Fixes #0000 (<=== Add issue number here)

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [ ] PR is descriptively titled 📑 and links the original issue above 🔗
* [ ] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [ ] code is in uniquely-named feature branch and has no merge conflicts 📁
* [ ] screenshots/GIFs are attached 📎 in case of UI updation
* [ ] ask `@publiclab/reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software 

Thanks!

Tlazypanda commented:
Hey @cesswairimu @jywarren  I think the cache was corrupted because we moved to the bundler version 2. Also the travis had a recent addition to their docs on how to handle moving to bundler 2(https://docs.travis-ci.com/user/languages/ruby/#bundler-20). So in this pr I moved the gem lockfile to version 2 and also updated the rubygems to v3. Then had to remove the cache folder to fresh install all gems. After we get this merged we can open a new pr to remove the line for removing the cache since the new cache folder will be up-to-date. I am not sure if this is the right solution but it might be a temporary fix :sweat_smile:

cesswairimu commented:
@Tlazypanda you are a life saver :100: ....thanks :tada:

jywarren commented on .travis.yml:
```diff
@@ -32,20 +32,24 @@ before_install:
   - sudo unzip chromedriver_linux64.zip -d /usr/local/bin/
   - sudo chmod +x /usr/local/bin/chromedriver
   - google-chrome-stable --headless --disable-gpu --remote-debugging-port=9222 http://localhost &
+  - yes | gem update --system --force
```
Hmm, could this have caused Jenkins to fail?

jywarren commented:
Hi all! It seems this commit gave Jenkins some trouble on stable: https://jenkins.laboratoriopublico.org/job/Plots-Stable/2050/console

I marked a line above that could be responsible, but not sure!

Aha - looks like Bundler 2 is the issue!

```
You must use Bundler 2 or greater with this lockfile.
Service 'mailman' failed to build: The command '/bin/sh -c bundle update --bundler &&     bundle install --jobs=4' returned a non-zero code: 20
Makefile:2: recipe for target 'build' failed
make: *** [build] Error 1
Build step 'Execute shell' marked build as failure
Finished: FAILURE
```

Tlazypanda commented:
@jywarren Yes, we migrated to bundler version 2 since while debugging the travis script bundler -v was showing as 2 but the lockfile wasn't updated and was still at 1 hence I changed it and rebuild the gems for the travis script to run :sweat_smile: 

> Hey @cesswairimu @jywarren I think the cache was corrupted because we moved to the bundler version 2. Also the travis had a recent addition to their docs on how to handle moving to bundler 2(https://docs.travis-ci.com/user/languages/ruby/#bundler-20). So in this pr I moved the gem lockfile to version 2 and also updated the rubygems to v3. Then had to remove the cache folder to fresh install all gems. After we get this merged we can open a new pr to remove the line for removing the cache since the new cache folder will be up-to-date. I am not sure if this is the right solution but it might be a temporary fix

jywarren commented:
Hmm. How is Bundler 2 installed? I think we may need to make a corresponding change to the Jenkins or Docker configurations. Could it be a change on this line? 

https://github.com/publiclab/plots2/blob/c60aaabb56c018962c76abf3f5f2d114f8492d15/Dockerfile#L42

Or maybe something needed in one of these files? https://github.com/publiclab/plots2/blob/master/containers/docker-compose-stable.yml

How did we migrate to bundler 2 -- via dependabot? Thank you for your help, and esp. for getting Travis running again!!!

Tlazypanda commented:
Hey @jywarren can we refer the instructions here for docker this? https://stackoverflow.com/questions/53231667/bundler-you-must-use-bundler-2-or-greater-with-this-lockfile

jywarren commented:
Aha, indeed:

```
For Docker projects in Ruby on Rails

If you're experiencing this issue when trying to build your application using Docker, simply do this:

Delete the Gemfile.lock file

Please don't create it again by running bundle install.

Run your docker build or docker-compose build command as appropriate to build your project.

This will re-create the Gemfile.lock file and setup the appropriate version of bundler necessary for your project in the Gemfile.lock file.

N/B: As a side note, you can also add this environment variable to your Dockerfile with the Bundler version defined.

ENV BUNDLER_VERSION=2.1.4
```

jywarren commented:
How to best do this in the Jenkins env?

jywarren commented:
Ah!!! @cesswairimu has a PR already -- https://github.com/publiclab/plots2/pull/8091 -- amazing, thanks Cess!

jywarren commented:
And thank you @Tlazypanda for looking into this!

Tlazypanda commented:
Ohh awesome thanks @cesswairimu @jywarren I didn't take the `stable` into consideration while solving the travis fix :sweat_smile:  Thanks for your patience on this one :heart:
